src/Umbraco.Cms.ManagementApi/Controllers/Document/RecycleBin/DocumentRecycleBinControllerBase.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Models.Entities;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.ManagementApi.Filters;
+using Umbraco.Cms.ManagementApi.ViewModels.RecycleBin;
+using Umbraco.New.Cms.Web.Common.Routing;
+
+namespace Umbraco.Cms.ManagementApi.Controllers.Document.RecycleBin;
+
+[ApiVersion("1.0")]
+[ApiController]
+[VersionedApiBackOfficeRoute($"{Constants.UdiEntityType.Document}/recycle-bin")]
+[RequireDocumentTreeRootAccess] // NOTE: this is pending evaluation for new backoffice
+public class DocumentRecycleBinControllerBase : RecycleBinControllerBase<RecycleBinItemViewModel>
+{
+    public DocumentRecycleBinControllerBase(IEntityService entityService)
+        : base(entityService)
+    {
+    }
+
+    protected override UmbracoObjectTypes ItemObjectType => UmbracoObjectTypes.Document;
+
+    protected override RecycleBinItemViewModel MapRecycleBinViewModel(Guid? parentKey, IEntitySlim entity)
+    {
+        RecycleBinItemViewModel viewModel = base.MapRecycleBinViewModel(parentKey, entity);
+
+        if (entity is IDocumentEntitySlim documentEntitySlim)
+        {
+            viewModel.Icon = documentEntitySlim.ContentTypeIcon ?? viewModel.Icon;
+        }
+
+        return viewModel;
+    }
```

Comment:
Had a bit of a rollercoaster ride of opinions about this mapping pattern. Normally I'd say that it shouldn't be the responsibility of the controller to know how to map to different entities, but just ensure that it gets done. 

But I can't deny how awesomely simple this makes it to implement a new tree controller, so I think I've landed on "Worth it" 🤣